### PR TITLE
Improve documentation of kdb.c with suggestions from API review

### PR DESCRIPTION
**Pull request by lawli3t**
## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: src/libs/elektra/kdb.c [hunk 1]
```diff
@@ -371,23 +371,23 @@ static int ensureContract (KDB * handle, const KeySet * contract, Key * parentKe
...
 /**
- * @brief Opens the session with the Key database.
+ * Opens the session with the Key database.
  *
  * @pre errorKey must be a valid key, e.g. created with keyNew()
  *
- * The method will bootstrap itself the following way.
+ * You must always call this method before retrieving or committing any
+ * keys to the database. In the end of the program,
+ * after using the key database, you must not forget to kdbClose().
```
Comment (mpranj) [inline src/libs/elektra/kdb.c:380]:
```suggestion
 * keys to the database. At the end of a program, after using the Key database (KDB),
 * you must not forget to call kdbClose() to free resources.
```

File: src/libs/elektra/kdb.c [hunk 2]
```diff
@@ -562,12 +565,11 @@ KDB * kdbOpen (const KeySet * contract, Key * errorKey)
  * Closes the session with the Key database.
  *
  * @pre The handle must be a valid handle as returned from kdbOpen()
- *
  * @pre errorKey must be a valid key, e.g. created with keyNew()
  *
  * This is the counterpart of kdbOpen().
  *
- * You must call this method when you finished your affairs with the key
+ * You must call this method when you are finished working with the key
```
Comment (mpranj) [inline src/libs/elektra/kdb.c:572]: Sometimes Key is uppercase, sometimes lowercase.

```suggestion
 * You must call this method when you are finished working with the Key
```

**Review by mpranj — APPROVED** [review]
Great work!

I added some suggestions. Please fix CI.

**Comment by mpranj** [comment]
Thank you so much!

**Comment by markus2330** [comment]
Amazing work :sparkling_heart:

**Pull request merged**